news-bundle/src/Resources/contao/modules/ModuleNewsArchive.php

```diff
@@ -189,51 +192,107 @@ protected function compile()
 			}
 		}
 
+		// Get the news items
+		$objArticles = $this->fetchItems($this->news_archives, $intBegin, $intEnd, $limit, $offset);
+
+		// Add the articles
+		if ($objArticles !== null)
+		{
+			$this->Template->articles = $this->parseArticles($objArticles);
+		}
+
+		$this->Template->headline = trim($this->headline);
+		$this->Template->back = $GLOBALS['TL_LANG']['MSC']['goBack'];
+		$this->Template->empty = $GLOBALS['TL_LANG']['MSC']['empty'];
+	}
+
+	/**
+	 *  Count the total matching items
+	 *
+	 * @param array $newsArchives
+	 * @param integer$intBegin
+	 * @param integer$intEnd
+	 *
+	 * @return int|null
+	 */
+	protected function countItems($newsArchives, $intBegin, $intEnd)
+	{
+		// HOOK: add custom logic
+		if (isset($GLOBALS['TL_HOOKS']['newsArchiveCountItems']) && \is_array($GLOBALS['TL_HOOKS']['newsArchiveCountItems']))
+		{
+			foreach ($GLOBALS['TL_HOOKS']['newsArchiveCountItems'] as $callback)
+			{
+				if (($intResult = System::importStatic($callback[0])->{$callback[1]}($newsArchives, $intBegin, $intEnd, $this)) === false)
+				{
+					continue;
+				}
+
+				if (\is_int($intResult))
+				{
+					return $intResult;
+				}
+			}
+		}
+
+		return NewsModel::countPublishedFromToByPids($intBegin, $intEnd, $newsArchives);
+	}
+
+	/**
+	 * Fetch the matching items
+	 *
+	 * @param array   $newsArchives
+	 * @param integer $intBegin
+	 * @param integer $intEnd
+	 * @param integer $limit
+	 * @param integer $offset
+	 *
+	 * @return Collection|NewsModel|NewsModel[]|null
+	 */
+	protected function fetchItems($newsArchives, $intBegin, $intEnd, $limit, $offset)
+	{
+		// HOOK: add custom logic
+		if (isset($GLOBALS['TL_HOOKS']['newsListFetchItems']) && \is_array($GLOBALS['TL_HOOKS']['newsListFetchItems']))
+		{
+			foreach ($GLOBALS['TL_HOOKS']['newsListFetchItems'] as $callback)
+			{
+				if (($objCollection = System::importStatic($callback[0])->{$callback[1]}($newsArchives, $intBegin, $intEnd, $limit, $offset, $this)) === false)
+				{
+					continue;
+				}
+
+				if ($objCollection === null || $objCollection instanceof Collection)
+				{
+					return $objCollection;
+				}
+			}
+		}
```

Comment:
The same hook should not be re-used for the news archive in my opinion.

Reply:
You can see via the supplied instance of the module whether you are in the archive module or in the list module. My thought here was that if, for example, you want to have a filter that serves both, then this does not have to be handled separately.

If the pull request should be accepted, be it as EVENT (5.x) or as HOOK. Then I could also separate them.

Reply:
> My thought here was that if, for example, you want to have a filter that serves both, then this does not have to be handled separately.

You don't have to handle it separately. You can still use one service to handle both cases.

Reply:
Okay then I will rename that to `newsArchiveFetchItems `. For now, I'll wait and see what the basic decision is on this PR.